lib/streamlit/elements/media.py

```diff
@@ -670,13 +720,32 @@ def marshall_audio(
         The time at which this element should stop playing
     loop: bool
         Whether the audio should loop playback.
+    autoplay : bool
+        Whether the audio should start playing automatically.
+        Browsers will not autoplay audio files if the user has not interacted with the page yet.
     """
 
     proto.start_time = start_time
     if end_time is not None:
         proto.end_time = end_time
     proto.loop = loop
 
+    if autoplay:
+        ctx = get_script_run_ctx()
+        proto.autoplay = autoplay
+        id = compute_widget_id(
+            "audio",
+            data=data,
+            mimetype=mimetype,
+            start_time=start_time,
+            sample_rate=sample_rate,
+            end_time=end_time,
+            loop=loop,
+            autoplay=autoplay,
+            page=ctx.page_script_hash if ctx else None,
+        )
+        proto.id = id
```

Comment:
I think it would be better to move this part to the end of the function and use `url=proto.url` instead of `data=data`. The reason is that the `data` here might not allow us a stable deterministic calculation of the widget id (or there is a higher risk that it doesn't work as expected), while the `proto.url` as a string definitely will. See the comment in `comput_widget_id`:

> Only stable, deterministic values should be used to compute widget ids. Using
    nondeterministic values as inputs can cause the resulting widget id to
    change between runs. 

And I believe the same data within the same session will always result in the same URL (but that might be something to confirm).

Reply:
I was in the process of pushing a commit with a serialization function I wrote to convert data into a string representation. But using `url` instead of `data` is much cleaner. Thanks for the idea! Fixed on 292bae7